### PR TITLE
Initialize SimulatorAccess object for compositing material models.

### PR DESCRIPTION
**Pull request by bangerth**
This fixes the fact that compositing material models (specifically the depth dependent
and averaging material models) held pointers to base models that may need to access
the simulator, but never initialized these base models' pointers to the Simulator
object. This patch fixes this by intercepting the initializing call.

### CODE REVIEW

**Comment by bangerth** [comment]
Let's give @Shangxin-Liu a chance to comment whether this works for him before merging.

**Comment by Shangxin-Liu** [comment]
I've put the change to the public version of ASPECT on our cluster @bangerth. The compilation has passed. However, when I make the compositing material model test with the depth-dependent and simple compressible model, the segmentation fault still appears:

[hs002:03064] *** Process received signal ***
[hs002:03064] Signal: Segmentation fault (11)
[hs002:03064] Signal code: Address not mapped (1)
[hs002:03064] Failing at address: 0x40
[hs002:03064] [ 0] /lib64/libpthread.so.0(+0xf500) [0x7fe0f31dd500]
[hs002:03064] [ 1] /home/shangxin/aspect_public/build/aspect(_ZNK6aspect15SimulatorAccessILi3EE20get_surface_pressureEv+0x4) [0x87a6a4]
[hs002:03064] [ 2] /home/shangxin/aspect_public/build/aspect(_ZNK6aspect13MaterialModel18SimpleCompressibleILi3EE8evaluateERKNS0_19MaterialModelInputsILi3EEERNS0_20MaterialModelOutputsILi3EEE+0xcd) [0x8cf46d]
[hs002:03064] [ 3] /home/shangxin/aspect_public/build/aspect(_ZNK6aspect13MaterialModel14DepthDependentILi3EE8evaluateERKNS0_19MaterialModelInputsILi3EEERNS0_20MaterialModelOutputsILi3EEE+0x24) [0x8dcfe4]
[hs002:03064] [ 4] /home/shangxin/aspect_public/build/aspect(_ZN6aspect19AdiabaticConditions14InitialProfileILi3EE10initializeEv+0x48b) [0x6db66b]
[hs002:03064] [ 5] /home/shangxin/aspect_public/build/aspect(_ZN6aspect9SimulatorILi3EEC2EP19ompi_communicator_tRN6dealii16ParameterHandlerE+0x19d1) [0x82d991]
[hs002:03064] [ 6] /home/shangxin/aspect_public/build/aspect(main+0x468) [0x6d77e8]
[hs002:03064] [ 7] /lib64/libc.so.6(__libc_start_main+0xfd) [0x7fe0f2e5acdd]
[hs002:03064] [ 8] /home/shangxin/aspect_public/build/aspect() [0x6d0d99]
[hs002:03064] *** End of error message ***
--------------------------------------------------------------------------

**Comment by bangerth** [comment]
OK. I'll investigate tomorrow.

**Comment by Shangxin-Liu** [comment]
Thanks. If you need my prm file to make the tests on your cluster, I can send you my depth-dependent viscosity file.

**Comment by bangerth** [comment]
Yes, either send it to me or copy it here.

File: doc/modules/changes.h [hunk 1]
```diff
@@ -6,9 +6,15 @@
  *
  *
  * <ol>
- * <li> New: The tolerance of the preconditioners of the A and S block 
- * are now available as parameters in the prm file. There is now also 
- * a section added to the manual on how to use these parameters to 
+ * <li> Fixed: Whenever the base models used by either the "depth dependent"
+ * or "averaging" material models depended on anything that requires accessing
+ * the simulator, then this led segmentation faults. This is now fixed.
```
Comment (tjhei) [inline doc/modules/changes.h]: "led to"?

**Comment by tjhei** [comment]
looks good, please merge yourself after fixing the typo.

File: doc/modules/changes.h (hunk 1 above)
Comment (Shangxin-Liu) [inline doc/modules/changes.h]: Hi Wolfgang and Timo,

The segmentation fault problem has been solved. Thanks!

Shangxin

**Pull request merged**